doc/examples/features_detection/plot_scattering_classification.py

```diff
@@ -0,0 +1,259 @@
+see the example *Scattering Features* or check [1]_ and [2]_.
+
+ **Supervised Learning: Image Classification**
+The first task in supervised image classification is to collect a
+set of images tagged with a class. In case of the MNIST
+database we have pictures of digits, and the class is the
+corresponding number. Then, for every image
+we compute a feature vector which, in our case, is the
+scattering vector. Once we have the complete
+database, we 'train' a classifier, meaning, we find the
+best parameters that allow to correctly classify the images.
+A widely used classifier is Support Vector Machine (SVM),
+which searches the high dimensional plane
+that allows to better separate each pair of classes,
+see [3]_ for more information.
+Here, we show how to reproduce the results obtained
+with the scattering transform, using SVM (with Gaussian kernels)
+first presented in the following papers:
+
+- **MNIST database** :
+ The results were first presented in [1]_ Table 4. Using this
+ implementation and a 10000 image test set, we were able to obtain, the following
+ error results :
+            +-------------------------+---------+
+            |num images training set  | % error |
+            +-------------------------+---------+
+            |                    300  |  5.92   |
+            +-------------------------+---------+
+            |                  1000   |  2.7    |
+            +-------------------------+---------+
+            |                  2000   |  2      |
+            +-------------------------+---------+
+            |                  5000   |  1.32   |
+            +-------------------------+---------+
+            |                 10000   |  0.96   |
+            +-------------------------+---------+
+            |                 20000   |  0.82   |
+            +-------------------------+---------+
+            |                 40000   |  0.56   |
+            +-------------------------+---------+
+            |                 60000   |  0.51   |
+            +-------------------------+---------+
+
+- **CIFAR10 database**:
+ This is a very challenging database with
+ 10 classes and 32x32 color images. The code presented here can
+ reproduce the results obtained in Table 1 (Trans., order 1)
+ of [2]_, more specifically it can obtain 73.6% success for m=1
+ and 40000 images in the training set.
+
+**Warning**: For integration reasons, the code uses very few images
+in the datasets. To obtain full performance, you need
+to change the variable num_images to 10000 at least.
+
+.. [1] Bruna, J., Mallat, S. 'Invariant Scattering Convolutional Networks'.
+IEEE TPAMI, 2012.
+
+.. [2] Oyallon, E. et Mallat, S. 'Deep Roto-translation Scattering
+ for Object Classification'. CVPR 2015
+
+.. [3] https://en.wikipedia.org/wiki/Support_vector_machine
+"""
+#######################
+# MNIST database
+# Modules, MNIST database and functions for scattering feature generation
+import time as time
+from keras.datasets import mnist
```

Comment:
Because we don't want that dependency. It means people would have to install the package to run the example, which implies the installation of a lot of other stuff.

But I'd let a core contributor comment on this -- maybe I am mistaken in the policy and for examples it's OK? In general one likes to keep dependencies at a minimum.